src/stream.js

```diff
@@ -75,41 +112,49 @@ module.exports = ({ id, name, send, onEnd = () => {}, type = 'initiator', maxMsg
       onSinkEnd(err)
```

Comment:
The problems seems here, both in reset and abort. They call `onSinkEnd` and do the abort. This will cause the sink catch to be triggered and a second `onSinkEnd` happens in the end. 

Removing `onSinkEnd` in abort and reset will not work as if the sink did not start the onEnd will not be triggered. So, we will likely need to improve the logic in the catch function of sink to not do the following on abort/reset:

```js
stream.source.end(err)
return onSinkEnd(err)
```

Without the previous change, this was not problematic somehow. I think it is related to something now happening in a different event loop.

What do you think to be the best approach here?

Reply:
I'm not entirely sure I follow. Both `onSourceEnd` and `onSinkEnd` are guarded from running twice, so both of these shouldn't do anything if ran a second time, right?

Reply:
Ah I see, I guess it's because in `abort`, `onEnd` is called before the `Types.Reset` message was sent. And with the new `_send` it can't be, and apparently this makes everything hang. Maybe a bug in some other software, I guess there should be a timeout somewhere.

I'll see tomorrow if I can clean this up a bit, but at least it works now.